ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/utils.py

```diff
@@ -137,6 +137,11 @@ def _get_l3_interface_cfg(self, l3_interface: dict) -> dict | None:
         Returns structured_configuration for one L3 interface
         """
         interface_name = get(l3_interface, "name", required=True, org_key=f"<node_type_key>...[node={self.shared_utils.hostname}].l3_interfaces[].name]")
+        if "." in interface_name:
+            iface_type = "l3dot1q"
+            encapsulation = get(l3_interface, "encapsulation_dot1q_vlan", default=interface_name.split(".")[-1])
```

Comment:
Encapsulation should be an integer.
```suggestion
            encapsulation = int(get(l3_interface, "encapsulation_dot1q_vlan", default=interface_name.split(".")[-1]))
```